RULES/cc/toolchain.go

```diff
@@ -97,6 +114,23 @@ func (gcc GccToolchain) EmbeddedBlob(out core.OutPath, src core.Path) string {
 		src)
 }
 
+// RawBinary strips ELF metadata to create a raw binary image
+func (gcc GccToolchain) RawBinary(out core.Path, elfSrc core.Path) string {
```

Comment:
out should be OutPath

Reply:
ok

Reply:
Somehow I thought you wanted to the name, not the type, now I understand. Thanks for pointing it out.

Reply:
Yes, that is what I meant. The `BuildStep` enforces that you can only use `OutPath` for outputs. Unfortunately we cannot do the same thing for commands (since they are just strings and how the output of a command is specified depends on the command), but at least we can force users of the toolchain to do the right thing.